src/FitterBase.cc

```diff
@@ -45,15 +45,15 @@ Chi2Statistic FitterBase::computeChi2() const {
     accumulateStatRefStars(chi2);
     // chi2.ndof contains the number of squares.
     // So subtract the number of parameters.
-    chi2.ndof -= _nParTot;
+    chi2.ndof -= _nTotal;
     return chi2;
 }
 
 std::size_t FitterBase::findOutliers(double nSigmaCut, MeasuredStarList &msOutliers,
                                      FittedStarList &fsOutliers) const {
     // collect chi2 contributions
     Chi2List chi2List;
-    chi2List.reserve(_nMeasuredStars + _associations->refStarList.size());
+    chi2List.reserve(_associations->refStarList.size());  // TODO: do we need to reserve anything?
```

Comment:
`_nMeasuredStars` was always zero (it must have been a placeholder for something, but was never implemented), so it wasn't doing anything here.

I've added `Associations._maxMeasuredStars`, calculated at initialization, so we can reserve a maximally-sized vector. It'll be bigger than necessary after outlier removal, but I think that's not a big problem, since this chi2 list is a tiny part of the memory footprint (it holds a pointer and a chi2 value for each star).